arangod/Cluster/HeartbeatThread.cpp

```diff
@@ -360,7 +358,7 @@ void HeartbeatThread::runDBServer() {
         break;
       }
 
-      double remain = interval - (TRI_microtime() - start);
+      auto remain = _interval - (std::chrono::steady_clock::now() - start);
```

Comment:
can `remain` get negative here? can it actually store negative values?

Reply:
is checked later here